hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/BaseHoodieWriteClient.java

```diff
@@ -244,6 +242,7 @@ public boolean commitStats(String instantTime, List<HoodieWriteStat> stats, Opti
     // We don't want to fail the commit if hoodie.fail.writes.on.inline.table.service.exception is false. We catch warn if false
     try {
       // do this outside of lock since compaction, clustering can be time taking and we don't need a lock for the entire execution period
+      postCommit(table, metadata, instantTime, extraMetadata, false);
```

Comment:
there is much more to this. this is skipping locks for inline cleaning. I will take over from here.